### PR TITLE
more robust code for avoiding the singularity in PML

### PR DESCRIPTION
**Pull request by xiezhinan**

### CODE REVIEW

**Comment by buildbot-princeton** [comment]
The files changed require a buildbot test. Buildbot has started.

**Comment by buildbot-princeton** [comment]
Your changes passed the buildbot test. See [build 400](http://specfem3d-buildbot.princeton.edu/builders/specfem3d_builder/builds/400).

File: src/generate_databases/pml_set_local_dampingcoeff.f90 [hunk 1]
```diff
@@ -1362,6 +1379,189 @@ subroutine pml_set_local_dampingcoeff(myrank,xstore,ystore,zstore)
     enddo
   enddo !ispec_CPML
 
+! for robust parameter separation of PML damping parameter
+  minimum_for_dx = minval(d_store_x)
+  minimum_for_dy = minval(d_store_y)
+  minimum_for_dz = minval(d_store_z)
+  maximum_for_dx = maxval(d_store_x)
+  maximum_for_dy = maxval(d_store_x)
```
Comment (danielpeter) [inline src/generate_databases/pml_set_local_dampingcoeff.f90]: should be:  maximum_for_dy = maxval(d_store_y)

File: src/generate_databases/pml_set_local_dampingcoeff.f90 [hunk 2]
```diff
@@ -1362,6 +1379,189 @@ subroutine pml_set_local_dampingcoeff(myrank,xstore,ystore,zstore)
...
+  maximum_for_dz = maxval(d_store_x)
```
Comment (danielpeter) [inline src/generate_databases/pml_set_local_dampingcoeff.f90]: should be: maximum_for_dz = maxval(d_store_z)

**Comment by buildbot-princeton** [comment]
You pushed to your local branch while your previous changes are waiting to be committed. We assume that you did not intend for these changes to be part of the currently opened pull-request and ignore those changes. If you want these changes to be committed, please open a new pull-request.

**Note to maintainers:** when you accept the original pull-request, the message from buildbot will say that it **closed** the pull-request, as opposed to the usual message that it **merged** the pull-request. In spite of this, the initial pull-request will be merged after you accept it.

File: src/generate_databases/pml_set_local_dampingcoeff.f90 (hunk 1 above)
Comment (xiezhinan) [inline src/generate_databases/pml_set_local_dampingcoeff.f90]: Dear Daniel,

Thank you so much.
I have retested the code and correct the error according to your instruction.

Thanks again,
Best regards,
Zhinan






At 2015-11-16 14:19:38, "daniel peter" <notifications@github.com> wrote:


In src/generate_databases/pml_set_local_dampingcoeff.f90:

> @@ -1362,6 +1379,189 @@ subroutine pml_set_local_dampingcoeff(myrank,xstore,ystore,zstore)
>      enddo
>    enddo !ispec_CPML
>  
> +! for robust parameter separation of PML damping parameter
> +  minimum_for_dx = minval(d_store_x)
> +  minimum_for_dy = minval(d_store_y)
> +  minimum_for_dz = minval(d_store_z)
> +  maximum_for_dx = maxval(d_store_x)
> +  maximum_for_dy = maxval(d_store_x)


should be: maximum_for_dy = maxval(d_store_y)

**Pull request merged**